### DISCUSSION TITLE
Applying noise estimating Kalman filters to time slotted networks

### DISCUSSION THREAD

**Discussion by chrysn** [pendulum-project/statime#548] [Ideas]
A common issue with embedded low-power networks is that they need to have well synchronized clocks across larger intervals of time: The better the sync, the longer a device can stay off the network and then still wake up in the precise time slot when its peers may have relevant data. (This is primarily relevant for devices reacting; devices sending on their own initiative generally have a peer that has an always-on receiver, because pairing two power constrained devices was [apparently shown to be too hard](https://youtu.be/VXk9QZTbmmo?feature=shared&t=1661)).

Could the estimated noise make filtering and adjusting low power devices' clocks so efficient that they could maintain a clock synchronized with the network with a higher precision (thus allowing shorter error margins or equivalently longer off times)?

If so, could the regular beacons from the local coordinator (of which the constrained device may not listen to all of them, eg. in some super-frame setups) suffice as clock sources? Or would the need to come with their own estimation of the sending node's jitter to be usable, lest two synchronization algorithms fight each other?

(This was originally a question [responding to the fediverse toot on the paper](https://chaos.social/@chrysn/113250148756529812); the Tweede golf response encourated turning this into a discussion here)

**Comment by davidv1992** [comment]
I have very little experience with these types of networked devices, so can't tell in detail, but possibly. It depends on how you synchronize right now and how well that already performs. I would encourage you to try it, the algorithm building blocks are available in the library and should be reasonably flexible.
